src/model.rs

```diff
@@ -119,6 +119,27 @@ pub struct Collections {
 }
 
 impl Collections {
+    /// Remove associated schedules with points of route
+    pub fn remove_points_of_route(&mut self) {
+        let vj_idxs: Vec<Idx<VehicleJourney>> =
+            self.vehicle_journeys.iter().map(|(idx, _)| idx).collect();
+        for vj_idx in vj_idxs {
+            let mut vj = self.vehicle_journeys.index_mut(vj_idx);
+
+            let is_point_of_route = |stop_time: &mut StopTime| -> bool {
+                stop_time.pickup_type == 3 || stop_time.drop_off_type == 3
+            };
+            let mut stop_time_idx = 0;
+            while stop_time_idx < vj.stop_times.len() {
+                if is_point_of_route(&mut vj.stop_times[stop_time_idx]) {
+                    vj.stop_times.remove(stop_time_idx);
+                } else {
+                    stop_time_idx += 1
+                }
+            }
```

Comment:
You might be able to use something like `.retain()` here instead of the entire `while` loop.

```rust
vj.stop_times.retain(|stop_time| !is_point_of_route(stop_time));
```